packages/kamino-sdk/src/utils/utils.ts

```diff
@@ -69,6 +69,27 @@ export function buildStrategyRebalanceParams(params: Array<Decimal>, rebalance_t
     buffer.writeUint16LE(params[1].toNumber(), 2);
     buffer.writeUint16LE(params[2].toNumber(), 4);
     buffer.writeUint16LE(params[3].toNumber(), 6);
+  } else if (rebalance_type.kind == RebalanceType.Drift.kind) {
+    buffer.writeInt32LE(params[0].toNumber());
+    buffer.writeInt32LE(params[1].toNumber(), 4);
+    buffer.writeInt32LE(params[2].toNumber(), 8);
+    buffer.writeBigUint64LE(BigInt(params[3].toString()), 12);
+    buffer.writeUint8(params[4].toNumber(), 20);
+  } else if (rebalance_type.kind == RebalanceType.TakeProfit.kind) {
+    writeBigUint128LE(buffer, BigInt(params[0].toString()), 0);
+    writeBigUint128LE(buffer, BigInt(params[1].toString()), 16);
+    buffer.writeUint8(params[2].toNumber(), 32);
+  } else if (rebalance_type.kind == RebalanceType.PeriodicRebalance.kind) {
+    buffer.writeBigUint64LE(BigInt(params[0].toString()), 0);
+    buffer.writeInt32LE(params[1].toNumber(), 8);
```

Comment:
Please check `config.ts` again from `yvaults` repository (and also check onchain), as there are some differences. For example this param should be `writeUInt16LE` and also the ones below are different. Expander strat is missing 1 more param.